### PR TITLE
multipart viewer class->fc

### PR DESCRIPTION
**Pull request by jackkav**
make a get request to https://uo7d3b.sse.codesandbox.io/

you should see two drop down boxes for each part.

I chose to simplify the transformations in the component by adding them to the part object.

multipart sandbox https://codesandbox.io/s/laughing-goodall-uo7d3b?file=/index.js

### CODE REVIEW

**Review by filfreire — APPROVED** [review]

File: packages/insomnia/src/ui/components/viewers/response-multipart-viewer.tsx [hunk 1]
```diff
@@ -158,157 +125,148 @@ export class ResponseMultipartViewer extends PureComponent<Props, State> {
...
+              <i className="fa fa-caret-down fa--skinny space-left" />
             </DropdownButton>
-            <DropdownItem onClick={this._handleViewHeaders}>
-              <i className="fa fa-list" /> View Headers
-            </DropdownItem>
-            <DropdownItem onClick={this._handleSaveAsFile}>
-              <i className="fa fa-save" /> Save as File
-            </DropdownItem>
+            {parts.map(part => (
+              <DropdownItem key={part.title} value={part} onClick={selectPart}>
```
Comment (gatzjames) [inline packages/insomnia/src/ui/components/viewers/response-multipart-viewer.tsx]: We could use part.id instead
```suggestion
              <DropdownItem key={part.id} value={part} onClick={selectPart}>
```

File: packages/insomnia/src/ui/components/viewers/response-multipart-viewer.tsx [hunk 2]
```diff
@@ -158,157 +125,148 @@ export class ResponseMultipartViewer extends PureComponent<Props, State> {
...
+          contentType={getContentTypeFromHeaders(selectedPart.headers, 'text/plain')}
+          disableHtmlPreviewJs={disableHtmlPreviewJs}
+          disablePreviewLinks={disablePreviewLinks}
+          download={download}
+          editorFontSize={editorFontSize}
+          error={null}
+          filter={filter}
+          filterHistory={filterHistory}
+          getBody={partBuffer}
+          key={`${responseId}::${selectedPart?.title}`}
```
Comment (gatzjames) [inline packages/insomnia/src/ui/components/viewers/response-multipart-viewer.tsx]:
```suggestion
          key={`${responseId}::${selectedPart?.id}`}
```

File: packages/insomnia/src/ui/components/viewers/response-multipart-viewer.tsx [hunk 3]
```diff
@@ -158,157 +125,148 @@ export class ResponseMultipartViewer extends PureComponent<Props, State> {
...
+          previewMode={PREVIEW_MODE_FRIENDLY}
+          responseId={`${responseId}[${selectedPart?.title}]`}
```
Comment (gatzjames) [inline packages/insomnia/src/ui/components/viewers/response-multipart-viewer.tsx]:
```suggestion
          responseId={`${responseId}[${selectedPart?.id}]`}
```

**Review by gatzjames — APPROVED** [review]
Looks and works great!

**Pull request merged**